pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -0,0 +1,530 @@
+		snitch = snitches[0]
+	} else {
+		newSnitch := dmsclient.NewSnitch(snitchName, dmsi.Spec.Tags, "15_minute", "basic")
+		logger.Info("Creating snitch Name: ")
+		snitch, err = dmsc.Create(newSnitch)
+		if err != nil {
+			return err
+		}
+	}
+
+	ReSnitches, err := dmsc.FindSnitchesByName(snitchName)
+	if err != nil {
+		return err
+	}
+
+	if len(ReSnitches) > 0 {
+		if ReSnitches[0].Status == "pending" {
+			logger.Info("Checking in Snitch ...")
+			// CheckIn snitch
+			err = dmsc.CheckIn(snitch)
+			if err != nil {
+				logger.Error(err, "Unable to check in deadman's snitch", "CheckInURL", snitch.CheckInURL)
+				return err
+			}
+		}
+	} else {
+		logger.Error(err, "Unable to get Snitch by name")
+		return err
+	}
+	logger.Info("Snitch created nothing to do here.... ")
+	return nil
+}
+
+//Create secret containing the snitch url
+func (r *ReconcileDeadmansSnitchIntegration) createSecret(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, dmsc dmsclient.Client, cd hivev1.ClusterDeployment) error {
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", cd.Name, "cluster-deployment.Namespace:", cd.Namespace)
+	dmsSecret := cd.Spec.ClusterName + "-" + dmsi.Spec.SnitchNamePostFix + "-" + "dms-secret"
+	logger.Info("Checking if secret already exits")
+	err := r.client.Get(context.TODO(),
+		types.NamespacedName{Name: dmsSecret, Namespace: cd.Namespace},
+		&corev1.Secret{})
+	if errors.IsNotFound(err) {
+		logger.Info("Secret not found creating secret")
+		snitchName := cd.Spec.ClusterName + "." + cd.Spec.BaseDomain + "-" + dmsi.Spec.SnitchNamePostFix
+		ReSnitches, err := dmsc.FindSnitchesByName(snitchName)
+
+		if err != nil {
+			return err
+		}
+		for _, CheckInURL := range ReSnitches {
+
+			newdmsSecret := newDMSSecret(cd.Namespace, dmsSecret, CheckInURL.CheckInURL)
+
+			// set the owner reference about the secret for gabage collection
+			if err := controllerutil.SetControllerReference(&cd, newdmsSecret, r.scheme); err != nil {
+				logger.Error(err, "Error setting controller reference on secret")
+				return err
+			}
+			// Create the secret
+			if err := r.client.Create(context.TODO(), newdmsSecret); err != nil {
+				logger.Error(err, "Failed to create secret")
+				return err
+			}
+
+		}
+	}
+	logger.Info("Secret created , nothing to do here...")
+	return nil
+}
+
+//creating the syncset which contain the secret with the snitch url
+func (r *ReconcileDeadmansSnitchIntegration) createSyncset(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, cd hivev1.ClusterDeployment) error {
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", cd.Name, "cluster-deployment.Namespace:", cd.Namespace)
+	dmsSecret := cd.Spec.ClusterName + "-" + dmsi.Spec.SnitchNamePostFix + "-" + "dms-secret"
+	ssName := cd.Spec.ClusterName + "-" + dmsi.Spec.SnitchNamePostFix + "-" + "dms-secret"
```

Comment:
Sure this can be changed would you think this would cause confusion tho?

Reply:
I think it would be less confusing :see_no_evil: but if you think it would be more confusing, then that's also fair. It's not a big issue either way :slightly_smiling_face: